services/system/Transact/src/RTransact.cpp

```diff
@@ -95,15 +107,34 @@ void RTransact::onBlock()
    auto commitNum  = stat->current.commitNum;
    auto reversible = WriteOnly{}.open<ReversibleBlocksTable>();
    reversible.put({.blockNum = stat->current.blockNum, .time = stat->current.time});
-   BlockTime irreversibleTime = {};
+   BlockTime                      irreversibleTime = {};
+   std::vector<psibase::BlockNum> irreversible;
    for (auto r : reversible.getIndex<0>())
    {
       if (r.blockNum > commitNum)
          break;
       irreversibleTime = r.time;
       if (r.blockNum < commitNum)
+      {
          reversible.remove(r);
+         irreversible.push_back(r.blockNum);
+      }
    }
+
+   // Send replies to all transactions in the block that are now irreversible
+   PSIBASE_SUBJECTIVE_TX
```

Comment:
It's better to avoid sending to a socket inside a subjective transaction, as it can result in duplicate sending.